### DISCUSSION TITLE
Lesson 6 : Constructor requires no arguments (MacOS Catalina 10.15.7)

### DISCUSSION THREAD

**Discussion by JamiDEV** [smartcontractkit/full-blockchain-solidity-course-py#356] [Q&A] [answered]
Another unknown error. I've done all the digging I could for a resolution, could not find one. I did copy the incorrect Fundme file and then copied the correct one. I looked through my Mockv3 file and everything seems correct. What constructor might this be? 
![Screen Shot 2021-11-13 at 7 47 27 PM](https://user-images.githubusercontent.com/53233466/141665570-395b653f-3fce-4699-abbe-7d30a54679a6.jpg)

**Comment by cromewar** [comment, marked as answer]
Hello @JamiDEV  it seems you are passing arguments to FundMe constructor (such as the price feed address) and the contract is not expecting them, your FundMe.sol must have a constructor like this: 

```Solidity
   constructor(address _priceFeed) public {
        priceFeed = AggregatorV3Interface(_priceFeed);
        owner = msg.sender;
    }
```

Note: remember to delete the build folder and compile again anytime you make changes to the contracts, otherwise brownie is going to take the last version.

**Reply by JamiDEV** [reply]
Ah okay, I'll make the adjustments, youre great brother. Thank you

**Reply by cromewar** [reply]
Please let us now if this was helpful, otherwise we can find another solution

**Reply by JamiDEV** [reply]
So it didnt work, I went ahead and just copied the FundMe.sol file in the code repository as a last-ditch effort, a whole new error. 


MacBook-Pro-2:brownie_fund_me mymac$ brownie run scripts/deploy.py
Brownie v1.16.4 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Launching 'ganache-cli --port 8545 --gasLimit 12000000 --accounts 10 --hardfork istanbul --mnemonic brownie'...
  File "brownie/_cli/run.py", line 49, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 53, in run
    module = _import_from_path(script)
  File "brownie/project/scripts.py", line 149, in _import_from_path
    _import_cache[import_str] = importlib.import_module(import_str)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "<frozen, line line, in in
  File "./scripts/deploy.py", line 1, in <module>
    from brownie import FundMe , MockV3Aggregator, network ,config
ImportError: cannot import name 'FundMe' from 'brownie' (/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/brownie/__init__.py)
Terminating local RPC client...
MacBook-Pro-2:brownie_fund_me mymac$

**Reply by cromewar** [reply]
It says it cannot import FundMe, so you would like to delete the build folder and compile again the project ;)

**Reply by JamiDEV** [reply]
in getting the same error after doing that, I've deleted the build folder several times. notice this, when I compile the FundME sol doc is missing. 


MacBook-Pro-2:brownie_fund_me mymac$ brownie compile
Brownie v1.16.4 - Python development framework for Ethereum

New compatible solc version available: 0.8.9
Compiling contracts...
  Solc version: 0.8.9
  Optimizer: Enabled  Runs: 200
  EVM Version: Istanbul
Generating build data...
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/AggregatorInterface
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/AggregatorV2V3Interface
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/AggregatorV3Interface
 - MockV3Aggregator

Project has been compiled. Build artifacts saved at /Users/mymac/brownie_fund_me/build/contracts
MacBook-Pro-2:brownie_fund_me mymac$

**Reply by JamiDEV** [reply]
Also, I've noticed the constructor method is having issues, this is frustrating. It started showing with soliidty 0.6.6, fixes online said to change the version and i did 


![Screen Shot 2021-11-22 at 9 40 11 PM](https://user-images.githubusercontent.com/53233466/142972238-22fc10ed-9695-414a-8592-847518aa306b.png)
.

**Reply by cromewar** [reply]
There must be something messing up with your installation, i suggest you to reinstall eth-brownie, but before doing that your would like to delete `.bronwie` folder on for home directory, as it is a hidden folder don't forget to press the Command + Shift + . (period) keys at the same time, 

After deleting brownie try installing again. I hope this might help.

**Reply by JamiDEV** [reply]
You gotta be kidding me, it just started happening out of nowhere--- ugh this becoming unbearable, but thank you for your help so far brother.

**Reply by cromewar** [reply]
Please let us know if reinstalling was helpful for you.

**Reply by JamiDEV** [reply]
I fixed it, looks the fundme sol updated contract wasn't working because it's missing the contract method
![Screen Shot 2021-11-22 at 10 34 28 PM](https://user-images.githubusercontent.com/53233466/142975809-69e00e7d-9b4f-4532-b35d-b0004ef1bd87.png)
 
Im still getting errors for the chainlink imports

![Screen Shot 2021-11-22 at 10 35 42 PM](https://user-images.githubusercontent.com/53233466/142975877-86c6e3ed-3cfb-4cf7-bd3a-5d2a85d9939b.png)

but the Fund me contract still deployed. 

![Screen Shot 2021-11-22 at 10 36 28 PM](https://user-images.githubusercontent.com/53233466/142975920-19a84844-6200-49b7-89d8-857991641c2c.png)

 
Not sure if the Updated FundMe.sol is doing this for anyone else but it is for me.

**Reply by cromewar** [reply]
I'm glad you solved your problem, to get riff off linting issues with chainlink, please refer to [this](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md) link you should add them to vscode.

I'll close this thread now, please feel free to make any other questions.
